### PR TITLE
Huggingface support and action added

### PR DESCRIPTION
**Pull request by John-Peters-UW**

### CODE REVIEW

**Comment by John-Peters-UW** [comment]
Action failure is fine, see: https://github.com/gitter-lab/metl-pretrained/actions/runs/10478100258

I moved the ref to main so when we merge this, it will run on pushes/pr to main, however that breaks it for this PR

File: huggingface/combine_files.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        import_name = line.split('as')[-1].strip()
+        code = code.replace(f'{import_name}.', '')
+
+    huggingface_import = 'from transformers import PretrainedConfig, PreTrainedModel'
+    delimiter = '$>'
+
+    with open('./huggingface/huggingface_code.py', 'r') as f:
+        contents = f.read()
+        delim_location = contents.find(delimiter)
+        cut_contents = contents[delim_location+len(delimiter):]
```
Comment (agitter) [inline huggingface/combine_files.py:43]: Could you add some comments about what is going on here? My understanding is that we already have a version of the combined code from before and and now want to grab the first few lines of that when we automatically make an updated verison.

File: huggingface/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+huggingface-hub==0.23.4
+transformers==4.42.4
+numpy>=1.23.2
```
Comment (agitter) [inline huggingface/requirements.txt:3]: Is NumPy 2 going to cause issues here as well?

**Review by agitter — COMMENTED** [review]
Not many comments on this, it looks good already for the most part.

It would also help to have a couple sentence readme in the `huggingface` subdirectory explaining what this is all doing at a high level.

File: huggingface/requirements.txt (hunk 2 above)
Comment (John-Peters-UW) [reply huggingface/requirements.txt:3]: Downloading numpy-2.1.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.3 MB)

Guess not. It installs 2.1.0 as of right now and that worked fine in the linked action run that worked above.

**Comment by agitter** [comment]
I pushed a small cleanup commit in ec8493567a5ebb865cbc8492b9861207b57b5429 directly to `main`

**Pull request merged**